integration-tests/jest-reporter.js

```diff
@@ -0,0 +1,56 @@
+const _ = require('lodash/fp')
+const { parse } = require('path')
+const { createWriteStream, existsSync, mkdirSync } = require('fs')
+
+
+module.exports = class JestReporter {
+  constructor(_globalConfig, _options) {
+    this.logRootDir = process.env.LOG_DIR || '/tmp/test-results'
+  }
+
+  onTestStart(test) {
+    const { path } = test
+    console.info(`\n**  Running ${parse(path).name} at ${this.timeNow()}\n`)
+  }
+
+  onTestResult(_testRunConfig, testResult, _runResults) {
+    const { testFilePath, testResults } = testResult
+    const testName = parse(testFilePath).name
+    const hasFailure = testResults.some(({ status }) => status === 'failed')
+    const logDir = `${this.logRootDir}/${hasFailure ? 'failures' : 'successes'}`
+    const logFileName = `${logDir}/${testName}-${Date.now()}.log`
+    !existsSync(logDir) && mkdirSync(logDir, { recursive: true })
```

Comment:
Just be sure that I am not miss something here. Is there any reason why we would'n want to check existence of dir before create? Is this not a good practice anymore?

Searched in terra-ui repo, I found same code pattern [here](https://github.com/DataBiosphere/terra-ui/blob/ceb0005d2180773e32a588f3da74dbb886d5ca82/integration-tests/jest/jest-utils.js#L102).

Reply:
Hmm, `fs.mkdirSync` function is more "complex" than `fs.existsSync` function by the look of it.
https://github.com/nodejs/node/blob/c3aa86d6784af77121e5e98e75c6dc12e5bb39ec/lib/fs.js#L1346

Reply:
Two thoughts:

1. Since Node version 10, `mkdirSync` is said to handle the case where a directory exists (https://stackoverflow.com/a/71767385/2851999) so checking explicitly (via `!existsSync(logDir)`) seems unnecessary but I don't feel strongly if you'd like to be extra explicit.
2. I saw [recommendations for wrapping it in a `try-catch`](https://stackoverflow.com/a/24311711/2851999), like so:
```javascript
try {
  mkdirSync(logDir, { recursive: true })
} catch (e) {
  console.error(e)
}
```
What do you think?

Reply:
Sounds good. Added `try/catch` to handle `mkdirSync` error. Thank you.